common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/produce/ProduceProcessorManager.java

```diff
@@ -17,40 +17,89 @@
 
-import java.util.HashSet;
+import java.util.HashMap;
 import java.util.List;
-import java.util.Set;
+import java.util.Map;
 
 import javax.ws.rs.core.MediaType;
 
 import org.apache.servicecomb.foundation.common.RegisterManager;
 import org.apache.servicecomb.foundation.common.utils.SPIServiceUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ProduceProcessorManager extends RegisterManager<String, Map<String, ProduceProcessor>> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProduceProcessorManager.class);
 
-public final class ProduceProcessorManager extends RegisterManager<String, ProduceProcessor> {
   private static final List<ProduceProcessor> produceProcessor =
       SPIServiceUtils.getSortedService(ProduceProcessor.class);
 
   private static final String NAME = "produce processor mgr";
 
   public static final String DEFAULT_TYPE = MediaType.APPLICATION_JSON;
 
+  public static final String DEFAULT_SERIAL_CLASS = "servicecomb_default_class";
+
   public static final ProduceProcessorManager INSTANCE = new ProduceProcessorManager();
 
-  public static final ProduceProcessor JSON_PROCESSOR =
-      SPIServiceUtils.getTargetService(ProduceProcessor.class, ProduceJsonProcessor.class);
+  private Map<String, ProduceProcessor> jsonProcessorMap;
 
-  public static final ProduceProcessor PLAIN_PROCESSOR =
-      SPIServiceUtils.getTargetService(ProduceProcessor.class, ProduceTextPlainProcessor.class);
+  private Map<String, ProduceProcessor> plainProcessorMap;
 
-  public static final ProduceProcessor DEFAULT_PROCESSOR = JSON_PROCESSOR;
+  private Map<String, ProduceProcessor> defaultProcessorMap;
 
   private ProduceProcessorManager() {
     super(NAME);
-    Set<String> set = new HashSet<>();
     produceProcessor.forEach(processor -> {
-      if (set.add(processor.getName())) {
-        register(processor.getName(), processor);
-      }
+      Map<String, ProduceProcessor> prodProcessorMap = getObjMap()
+          .computeIfAbsent(processor.getName(), key -> new HashMap<>());
+      prodProcessorMap.putIfAbsent(processor.getSerializationView(), processor);
     });
+    jsonProcessorMap = ensureFindValue(MediaType.APPLICATION_JSON);
+    plainProcessorMap = ensureFindValue(MediaType.TEXT_PLAIN);
+    defaultProcessorMap = jsonProcessorMap;
+  }
+
+  public static ProduceProcessor cloneNewProduceProcessor(String acceptType, Class<?> serialViewClass,
+      Map<String, ProduceProcessor> produceViewMap) {
+    ProduceProcessor newInstance;
+    try {
+      newInstance = produceViewMap.get(DEFAULT_SERIAL_CLASS).getClass().newInstance();
+      newInstance.setSerializationView(serialViewClass);
+      return newInstance;
+    } catch (Throwable e) {
+      LOGGER.error(String.format("Failed to create produceProcessor with %s", acceptType), e);
+    }
+    return produceViewMap.get(DEFAULT_SERIAL_CLASS);
+  }
+
+  public Map<String, ProduceProcessor> getJsonProcessorMap() {
```

Comment:
It's not good to expose Map to callers. Maybe 
```
public ProduceProcessor getProduceProcessor(String name) 
```
is better. Following code are the same. 

And for setters, they seam not used, and can deleted.

Reply:
done